src/box/lua/connpool.lua

```diff
@@ -125,6 +125,42 @@ local function connect_to_candidates(candidates)
     return connected_candidates
 end
 
+local acquire_any_cond = nil
+local function acquire_any_candidate(instance_names, opts)
+    assert(type(opts) == 'table' and opts.mode)
+    assert(opts.mode == 'ro' or opts.mode == 'rw' or opts.mode =='alive')
+    local time_connect_end = clock.monotonic() + WATCHER_TIMEOUT
+
+    if not acquire_any_cond then
+        acquire_any_cond = fiber.cond()
+    end
```

Comment:
We have a similar `connection_mode_update_cond`, which is signaled by the `connect` function. This cond seems mostly same, but it is about a different phase. Can we implement them in a similar way: signal both from `connect`?

Also, it seems that `:wait_state()` that uses an internal cond may fit our needs for waiting `alive` (and even better -- it returns immediately on other terminal states). And we can reuse `connection_mode_update_cond` when waiting for a particular RO/RW mode.

What do you think?

Reply:
Unfortunately, it's not possible to use here without using extra fibers.

Reply:
OK regarding the `:wait_state()` proposal: it is per-connection, but we need one condvar. I missed it.

But the proposal regarding consistency with `connection_mode_update_cond` still actual. What do you think about it?

Reply:
Now, I see what you meant. I am sorry I wasn't able to understand it earlier. Applied it. Nice point! Thanks.

Reply:
Now the code reuses `connection_mode_update_cond`, but we only need the connected state here, AFAIR.

My comment was just about implementing these two condvars in a similar way, not about merging them into one.

Reply:
In fact, I don't mind this extra waiting for `mode`, it simplifies the code a bit.

I would just comment that the given `:wait()` call may wait for an extra time to obtain the mode even if it is not used later anyhow (for example, there are no dynamic filters).

Reply:
I think this is no longer related after https://github.com/tarantool/tarantool/pull/10438#discussion_r1862483309.